### DISCUSSION TITLE
Multiple frequencies reported by sensor Fineoffset-WN34

### DISCUSSION THREAD

**Discussion by FosterSM** [merbanan/rtl_433#2478] [Q&A] [answered]
Hi, I have a weather system that monitors a number of sensors from various manufacturers. I have found that the various sensors report slightly different tx frequencies and to get a reliable overall capture of data I have opted to use the "hop" feature from the rtl_433:

rtl_433 -H 7 -f 433910000 -f 433915000 -f 433920000 -f 433925000 -Y autolevel -F json -M bst -M level -d 0

This has significatly improved the reliability of my overall capture responses, the frequencies chosen and the hop value should be considered a work in progress and in no way cast in stone but at the moment I am happy with these values.

While analysing the responses to try and tune my frequency and hop selections I noticed that one of the sensors (Fineoffset-WN34) reports two frequencies freq1 and freq2 does anyone know the signifcance of this and in general can the returned freq values be relied upon in any way.
Thanks

**Comment by gdt** [comment]
Some things send FSK (frequency shift keying) and some send amplitude, pulse position, etc. that is only one frequency.  A decoded will report freq1/freq2 for FSK and just freq for things that are on one frequency.

What I would recommend is to save logs for a day in json and then graph for each sensor either freq or freq1/freq2 as it has it, and see how those line up and how they shfit over time.

It does not make sense to me that your strategy of moving slightly helps a lot.   One thing is that these dongles do not receive exactly at the chosen frequency.    So if you see a sensor's frequency varying and as it gets close to the configured freq you see dropouts, that makes sense.  But really I'd expect 433.92M to just work.

What kind of dongle, and for some particular sensor (let's choose one to talk about first, doesn't matter which) what frequencies are being reported?

**Reply by FosterSM** [reply]
Thank you for your reply:

Andoer® Mini Portable Digital USB 2.0 TV Stick DVB-T + DAB + FM RTL2832U + FC0012 is the dongle currently in use (I have others). 

At random one Temp/Humidity sensor I use is a Newentor this reports typically as: 

{"time" : "2023-04-09 20:13:18", "model" : "inFactory-TH", "id" : 69, "channel" : 2, "battery_ok" : 1, "temperature_F" : 68.100, "humidity" : 37, "mic" : "CRC", "mod" : "ASK", "freq" : 433.915, "rssi" : -0.683, "snr" : 10.989, "noise" : -11.671}

This sensor has reported the following frequencies over the last few hours (sorted to unique values), I was already tracking this particular sensor.

frequency 433.901000
frequency 433.902000
frequency 433.903000
frequency 433.904000
frequency 433.906000
frequency 433.908000
frequency 433.909000
frequency 433.911000
frequency 433.913000
frequency 433.914000
frequency 433.915000
frequency 433.916000
frequency 433.917000
frequency 433.918000
frequency 433.922000

Thanks

**Comment by gdt** [comment]
I have a dongle that looks just like that but is an old NooElec, and it is not very good compared to Blog v3 and NooElec SMART.

Keep in mind that the frequency change could be the dongle, or the sensor.   That doesn't look problematic, but it's a little surprising.   You are really going to need to analyze the data to see if you are missing decodes for long stretches of time, and if that happens near 433.920, or if over some times, or something else. 

Your snr values at not that high.   If the sensor is not near the receiver, that might be ok.

Have a look at #2438.

**Reply by FosterSM** [reply]
Prior to today the rtl_433 was using the default 433.920mhz and I was missing some decodes for long periods of time hence my investigation. I have a Blog V3 but I am awaiting a cable (SMA to F type) to connect to my aerial. I also have a dongle that is marked NooElec NESDR XTR which I could try but I am afraid to have too many changes going on at one time. I have a number of computers active to I would expect a lot of noise locally, my sensors are as close as 5mtrs to the aerial and nearly 50mtrs to the furthest and to be honest they all suffered drop outs from time to time no one is particulary affected. The effect of using the hop settings has made a big improvement  (no drop outs at all today) even though the frequency changes are small. I will collect the JSON data from all the sensors over tomorrow for analysis. #2438 looks like similar issues which I will take onboard.
Thanks

**Comment by gdt** [comment]
I advise picking one of your high-quality dongles and working with that first.  The NooElec NESDR XTR I have not tried but it looks very similar to ones I have, and they were great.  But, as you plan to collect data over a decent length of time and investigate, that sounds like the right thing to do.  I honestly do not understand why the hop strategy helps -- but that isn't proof it doesn't!

**Reply by FosterSM** [reply, marked as answer]
I have chosen the NooElec NESDR XTR dongle has this gives the best SNR (between 26 and 32). All the devices I have report a range of frequencies the NooElec reports an average frequency of 433.925523Mhz with a range from 433.885010Mhz to 433.9772992Mhz. I am using these figures as the basis for the "hop" spread in my application and I am very happy with the results, no dropouts in the last 24 hrs. Though I don't understand why this works I see no reason at the moment to change my strategy.
Thanks

**Comment by merbanan** [comment]
There is no perfect description of how the rtl-sdr dongle filter-chain works. But I have observed something similar. When I try to receive a specific sensor where I live I need to change the frequency to the higher end of the receiving band. If I use to low frequency it just does not work. I think there is a static signal that disturbs the signal demodulation. The conclusion is that if it doesn't work then adjust the frequency and see if it works better.

**Reply by FosterSM** [reply]
Yes I have found that using the hop strategy I have chosen I am receiving lots of transmissions for which I have no interest the system seems more sensitive than using the single fixed frequency making the signals I am interested in very stable reception wise.
Thanks
